spec/features/projects/life_cycle/overview_page/dialog/permission_spec.rb

```diff
@@ -0,0 +1,84 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require_relative "../shared_context"
+
+RSpec.describe "Edit project stages and gates on project overview page", :js, :with_cuprite,
+               with_flag: { stages_and_gates: true } do
+  include_context "with seeded projects and stages and gates"
+  let(:user) { create(:user) }
+  let(:overview_page) { Pages::Projects::Show.new(project) }
+  let(:permissions) { [] }
+
+  before do
+    allow(User).to receive(:current).and_return user
```

Comment:
You could also use the `current_user` [helper](https://github.com/opf/openproject/blob/1d73cab4a8932ccf8ecd40ab4a81f52ddb54313c/spec/support/authentication_helpers.rb#L104) here:

```ruby
current_user { user }
```